### ISSUE TITLE
Selecting Rinkeby, Kovan or Goerli chains throws an exception

### ISSUE THREAD

**Issue by assimovt** [ethereumjs/ethereumjs-monorepo#1552]
I am trying to load opcodes per chain and whenever I choose either `Rinkeby`, `Kovan` or `Goerli`, I get the following errors:

**For Rinkeby and Goerli:**
```
Uncaught Error: consensus (signature) validation only supported for poa clique algorithm
```

**For Kovan:**
```
bundle.js:2893 Uncaught (in promise) TypeError: state[0].toBuffer is not a function
```

I've put up a demo repo to show this in action, which you can find here: https://github.com/assimovt/ethereumjs-chain-bug

Could anybody help to resolve this issue? Thanks!

**Comment by ryanio** [comment]
I took a brief look at your code and you have to instantiate a new VM for each chain, you can't use `common.setChain()` like that.

**Comment by assimovt** [comment]
Now I've gotten rid of setChain and initializing every time, but it still fails the same way: https://github.com/assimovt/ethereumjs-chain-bug/commit/ea313025cc5fd3dcf83a9c2a917fcb1ef11e2177 🤔

**Comment by holgerd77** [comment]
@assimovt the first error for Rinkeby and Goerli is coming from the `Block` library. You need to initialize objects like your blocks (or txs) with the same Common you are using for the VM (at least that's my guess what's happening, haven't looked into your wider code) to operate with the same chain and HF settings.

Also, just as a note: if you are initializing e.g. the VM without a dedicated HF given, on the current library versions this is then initialized with `istanbul` (backwards-compatibility for non-breaking releases). Not sure if you want that.

Hope that helps!

**Comment by assimovt** [comment]
@holgerd77 Thanks for your answer, but I don't think I am following. I am not using `Block` and the only thing this does -  is follows the [example of the repo](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/common#usage) to initialize a VM for a given Common chain: https://github.com/assimovt/ethereumjs-chain-bug/blob/master/index.js#L13-L14

Do you happen to have an example of how it's done?

**Comment by acolytec3** [comment]
I'm going to take another look at this.  I think there might be something more that's going on when the `VM.init` call is made.

**Comment by acolytec3** [comment]
After some more research, there is indeed a bug in our code where we are incorrectly using the bn.js `toBuffer()` function instead of `toArrayLike()` that impacts usage of the `Blockchain` in browser contexts (and by extension the VM since it then depends on `BlockChain`).  Will have PR ready shortly to address this.

**Comment by holgerd77** [comment]
Ok, fix has been published along Blockchain `v5.5.1`, see #1567.
